File: wrappers/Javascript/PteroVM.js
const fs = require('fs');
const https = require('https');
const { exec } = require('child_process');

function main() {
    const url = "https://raw.githubusercontent.com/dxomg/vpsfreepterovm/main/PteroVM.sh";
    const destination = "PteroVM.sh";

    downloadFile(url, destination)
        .then(() => {
            // Set executable permission on downloaded file
            fs.chmodSync(destination, '755');
            
            // Run the downloaded file
            exec(`sh ./${destination}`, (error, stdout, stderr) => {
                if (error) {
                    console.error(`Error running script: ${error}`);
                    return;
                }
                console.log(stdout);
                
                // Remove the downloaded script after running
                fs.unlinkSync(destination);
            });
        })
        .catch((error) => {
            console.error(`Error downloading script: ${error}`);
        });
}

function downloadFile(url, destination) {
    return new Promise((resolve, reject) => {
        const file = fs.createWriteStream(destination);
        https.get(url, (response) => {
            response.pipe(file);
            file.on('finish', () => {
                file.close();
                resolve();
            });
        }).on('error', (error) => {
            fs.unlinkSync(destination);
            reject(error);
        });
    });
}

main();

File: wrappers/Python/PteroVM.py
import urllib.request
import os
import subprocess

def main():
    url = "https://raw.githubusercontent.com/dxomg/vpsfreepterovm/main/PteroVM.sh"
    destination = "PteroVM.sh"
    
    try:
        download_file(url, destination)
        
        # Set executable permission on downloaded file
        os.chmod(destination, 0o755)
        
        # Run the downloaded file
        subprocess.run(["sh", destination], check=True)
        
        # Remove the downloaded script after running
        os.remove(destination)
    except Exception as e:
        print(f"Error downloading or running script: {e}")
        raise

def download_file(url, destination):
    urllib.request.urlretrieve(url, destination)

if __name__ == "__main__":
    main()
